torch/_inductor/fx_passes/reinplace.py

```diff
@@ -0,0 +1,258 @@
+from collections import defaultdict, namedtuple
+import torch
+from torch._inductor.lowering import (
+    inplaceable_foreach_ops as inplaceable_foreach_ops_lowerings,
+)
+
+
+aten = torch.ops.aten
+InplaceableOp = namedtuple("InplaceableOp", ["inplace_op", "mutated_arg"])
```

Comment:
While we are doing this refactor, can we add typing info to these NamedTuples?

Reply:
Added in the top PR of the stack. I don't want to do any more that code movement in this PR.